feat(base): add support for Identifier class

nagajaideep opened the pull request:
## Pull Request description

<!-- Describe the purpose of your PR and the changes you have made. -->
This PR introduces support for the Identifier class in the AST system.
-Added the Identifier class to handle identifier expressions.
<!-- Which issue this PR aims to resolve or fix? E.g.:
Solve #168 
-->

## How to test these changes
-pre-commit run --all-files

<!-- Modify the options to suit your project. -->

## Pull Request checklists

This PR is a:

- [ ] bug-fix
- [x] new feature
- [ ] maintenance

About this PR:

- [x] it includes tests.
- [x] the tests are executed on CI.
- [ ] the tests generate log file(s) (path).
- [x] pre-commit hooks were executed locally.
- [ ] this PR requires a project documentation update.

Author's checklist:

- [ ] I have reviewed the changes and it contains no misspelling.
- [ ] The code is well commented, especially in the parts that contain more
      complexity.
- [ ] New and old tests passed locally.

## Additional information

<!-- Add any screenshot that helps to show the changes proposed -->

<!-- Add any other extra information that would help to understand the changes proposed by this PR -->

## Reviewer's checklist

```
## Reviewer's Checklist

- [ ] I managed to reproduce the problem locally from the `main` branch
- [ ] I managed to test the new changes locally
- [ ] I confirm that the issues mentioned were fixed/resolved .
```

nagajaideep commented:
solves this issue #168

nagajaideep commented:
@xmnlab hey issues is resolved i made identifier in base.py

xmnlab commented:
Hi @nagajaideep , thank you so much, apreciate that. Could you also add it to `__init__`, and add a test for that, please?

nagajaideep commented:
Sure will do it

nagajaideep commented:
Hey Ivan, good evening! I’ve implemented the tests for the Identifier class in the appropriate files and updated the PR description accordingly. Let me know if any further changes are needed.

xmnlab commented on tests/test_base.py:
```diff
@@ -69,3 +69,35 @@ def test_data_type() -> None:
+    loc = astx.SourceLocation(1, COLUMN_NUMBER)
+    ident_with_loc = astx.Identifier("var2", loc=loc)
+    assert ident_with_loc.value == "var2"
+    assert ident_with_loc.loc.line == 1
```
it is not necessary to test **loc**.
but I will not block this PR because of that.

xmnlab commented:
thank you @nagajaideep ! appreciate that.
